src/OrganisationUnitTree/OrganisationUnitNode.js

```diff
@@ -0,0 +1,129 @@
+import { Node } from '@dhis2/ui-core'
+import React from 'react'
+import propTypes from 'prop-types'
+import cx from 'classnames'
+
+import { orgUnitPathPropValidator, orgUnitIdPropValidator } from './propTypes'
+import { Label } from './Label'
+import { useIdFromPath } from './useIdFromPath'
+import { useOpenState } from './useOpenState'
+import { useOrgData } from './useOrgData'
+import { useChildren } from './useChildren'
+import { useHasSelectedDescendants } from './useHasSelectedDescendants'
+import { useIsUnitSelected } from './useIsUnitSelected'
+import { useOnUnitLoaded } from './useOnUnitLoaded'
+import { useIsHighlighted } from './useIsHighlighted'
+import { useContent } from './useContent'
```

Comment:
I'm not sure a class is more appropriate. I find classes even harder to read because I have to jump between content to actually get what's going on. But I get your point. I'd like to know what the others think about this as well (@dhis2/team-apps).

It only has one state: `open`. Everything else is more or less derived from props or a one-time effect (loading the data).